lib/kontrast/global_runner.rb

```diff
@@ -0,0 +1,121 @@
+require "yaml"
+require "net/http"
+
+module Kontrast
+    class GlobalRunner
+
+        def initialize
+            @page_runner = PageRunner.new
+            @api_endpont_runner = ApiEndpointRunner.new
+            @path = Kontrast.path
+            @current_node = 0
+        end
+
+        def run
+            # Make sure the local server is running
+            wait_for_server
+
+            # Assign nodes
+            if Kontrast.configuration.run_parallel
+                total_nodes = Kontrast.configuration.total_nodes
+                @current_node = Kontrast.configuration.current_node
+            else
+                # Override the config for local use
+                total_nodes = 1
+            end
+
+            # Run both runners
+            @page_runner.run(total_nodes, @current_node)
+            @api_endpont_runner.run(total_nodes, @current_node)
+
+            # Create manifest
+            create_manifest
+        end
+
+        # The manifest is a per-node .json file that is used to create the gallery
+        # without having to download all assets from S3 to the test environment
+        def create_manifest
```

Comment:
I kinda agree, but given the current architecture of the project, I can't really think of a better way.
It doesn't bother me that much to have it here as the manifest creation is currently a crucial part of the run process.